hudi-common/src/main/java/org/apache/hudi/avro/HoodieAvroReaderContext.java

```diff
@@ -91,10 +125,23 @@ public GenericRecord convertToAvroRecord(IndexedRecord record, Schema schema) {
 
   @Override
   public Option<HoodieRecordMerger> getRecordMerger(RecordMergeMode mergeMode, String mergeStrategyId, String mergeImplClasses) {
-    if (mergeMode == RecordMergeMode.CUSTOM) {
-      return customMerger;
+    switch (mergeMode) {
+      case EVENT_TIME_ORDERING:
+        return Option.of(new HoodiePreCombineAvroRecordMerger());
```

Comment:
Should a new merger `EventTimeAvroRecordMerger` be created in this case?  `HoodiePreCombineAvroRecordMerger` is only used for dedup and does not handle deletes.

Reply:
How does this work today in the paths that use merger? We should just use that I suppose. Is it the `HoodieAvroRecordMerger`?

Reply:
For the payload support, yes, `HoodieAvroRecordMerger` is used along with the payload class name.

Reply:
i.e., `HoodieAvroRecordMerger` with `DefaultHoodieRecordPayload` for event-time-based semantics.  Is the intention of this line to get rid of the payload class usage for `EVENT_TIME_ORDERING`, i.e., directly merging indexed records?

Reply:
Yes.  We discussed this offline; since the merger class is going to be refactored, we defer the implementation of `EventTimeRecordMerger` and use `HoodieAvroRecordMerger` for functional correctness.